src/H5VLpublic.h

```diff
@@ -49,6 +49,25 @@
  */
 #define H5_VOL_MAX 65535
 
+/* Flags to return from H5VLquery_optional API and 'opt_query' callbacks */
+/* Note: Operations which access multiple objects' data or metadata in a
+ *      container should be registered as file-level optional operations.
+ *      (e.g. "H5Dwrite_multi" takes a list of datasets to write data to, so
+ *      a VOL connector that implemented it should register it as an optional
+ *      file operation, and pass-through VOL connectors that are stacked above
+ *      the connector that registered it should assume that dataset elements
+ *      for _any_ dataset in the file could be written to)
+ */
+#define H5VL_OPT_QUERY_SUPPORTED       0x0001 /* VOL connector supports this operation */
```

Comment:
When wrappers are created, these may need to be added - at least to Java constants.